packages/lib/src/dependency-manager.ts

```diff
@@ -88,49 +125,241 @@ export class DependencyManager extends GirNSRegistry {
             : `import type ${namespace} from '${importPath}';`
     }
 
+    protected async parseGir(path: string) {
```

Comment:
Caching the XML in my experience tends to lead to memory issues because you're caching massive strings for each library currently being processed. I don't think there is a need to cache the XML anymore, is there?

Reply:
I have to parse the xml to extract the version number to compare the gir files for the copy command and so that I don't have to parse the xml again (before the real parsing starts) I cache it, but after that we could delete it again. Do you think that would make sense?

Reply:
oh yes, I think it was in the ejs templates, but it may be that async now also works in ejs, I can try it out

Reply:
I would cache the version number itself, caching the XML structure or parsed XML adds a lot of memory usage which will require increasing node's memory limits.

Reply:
Okay, I will implement it like this